src/category_theory/types.lean

```diff
@@ -182,5 +183,22 @@ def equiv_iso_iso {X Y : Sort u} : (X ≃ Y) ≅ (X ≅ Y) :=
   inv := λ i, i.to_equiv, }
 
 /-- equivalences (between types in the same universe) are the same as (equivalent to) isomorphisms of types -/
-def equiv_equiv_iso {X Y : Sort u} : (X ≃ Y) ≃ (X ≅ Y) :=
+-- We leave `X` and `Y` as explicit arguments here, because the coercions from `equiv` to a function won't fire without them.
+def equiv_equiv_iso (X Y : Sort u) : (X ≃ Y) ≃ (X ≅ Y) :=
 equiv_iso_iso.inv (equiv_iso_iso)
+
+@[simp] lemma equiv_iso_iso_hom {X Y : Sort u} (e : X ≃ Y) : equiv_iso_iso.hom e = e.to_iso := rfl
+@[simp] lemma equiv_iso_iso_inv {X Y : Sort u} (e : X ≅ Y) : equiv_iso_iso.inv e = e.to_equiv := rfl
+@[simp] lemma equiv_equiv_iso_hom {X Y : Sort u} (e : X ≃ Y) : (equiv_equiv_iso X Y) e = e.to_iso := rfl
+@[simp] lemma equiv_equiv_iso_inv {X Y : Sort u} (e : X ≅ Y) : (equiv_equiv_iso X Y).symm e = e.to_equiv := rfl
+
+namespace category_theory
+
+def Aut_mul_equiv_perm {α : Type u} : Aut α ≃* equiv.perm α :=
```

Comment:
You could do a similar cheat as with `equiv_equiv_iso`...
First define the thing as an iso in `Mon` or `Group` or whatever, and then deduce this one. It'll give you your auto_params (-;

Reply:
I thought about that, but then this file would have to import `algebra.category.Group`, or this content would have to move there, and my head started hurting.

Reply:
But we'll probably want the groupy version of `equiv_equiv_iso` somewhere at some point...

Reply:
Okay, I'm succumbing to feature creep and adding this now.

Reply:
Okay, all the multiplicative versions are here now as well. There was no point doing this without rebasing on top of @urkud's (lovely) work on concrete categories, so this PR will now have to wait until #1380 and #1438 have bene merged.